src/classes/UTIL_ACCT_Naming.cls

```diff
@@ -266,121 +283,90 @@ public class UTIL_ACCT_Naming {
         return finalAccountName;
     }
 
-    private static String chLToken = '{!';
-    private static String chRToken = '}';
-    private static Boolean setFNSpec = false;
 
     /*******************************************************************************************************
-    * @description Class that supports the parsing of Account name format set in the current org and Account model.
-    * NameSpec looks like: Prefix {LastName} {{FirstName}} Suffix
-    * firstNameSpec is a combo of fields and literals like: {Salutation} {FirstName} Family
-    * @return NameSpec.
+    * @description Given a NameSpec and a list of Contacts (assumed from a single household),
+    * return the formated Account name.
+    * This can be used for any of the household strings (name, formal greeting, informal greeting).
+    * @param cons is the list of Contacts
+    * @param ns the NameSpec class
+    * return String.
     */
-    private class NameSpec {
-        private String namePrefix { get; set; }
-        private String nameSuffix { get; set; }
-        private String firstNameSpec { get; set; }
-        private String fullNameSpec { get; set; }
-        private String andConnector { get; set; }
-        private string acctNameFormat { get; set; }
-        private String salutationSpec {get; set; }
-
-        /*******************************************************************************************************
-        * @description NameSpec Constructor
-        * @param accountNamingFormat is a string of the Account name format currently set in the org.
-        */
-        private NameSpec(String accountNamingFormat) {
-            String strNameSpec;
-
-            if (accountNamingFormat != NULL) {
-                strNameSpec = String.valueOf(accountNamingFormat);
-                parseNameSpec(strNameSpec);
-            }
-        }
-
-        /*******************************************************************************************************
-        * @description Given the strNameSpec string value, parse out its constituent parts, and sets them in the class.
-        * @param strNameSpec a string value of the Account name format. ie. {!FirstName} {!LastName} Family.
-        */
-        private void parseNameSpec(String strNameSpec) {
-            namePrefix = '';
-            nameSuffix = '';
-            firstNameSpec = '';
-            fullNameSpec = '';
-            andConnector = ' ' + Label.defaultNamingConnector + ' ';
-
-            if (strNameSpec == NULL) {
-                return;
-            }
+    private static String buildAccountName(List<Contact> cons, NameSpec ns) {
```

Comment:
Lots of changes from here to the end of the PR. I'd be easier to compare the before and after if the methods and inner class(es) were kept in the same order...

Reply:
I changed the order because the inner class was buried in amongst all the other methods of the class. Maybe it's just a nit-pick for me, but I don't like an entire class mixed in amongst all the methods so I moved it. The good news is I didn't have to change the inner class. That was working fine

Reply:
build account name did not change other than it calling a refactored method now called stringConNames where it used to call cleanFirstNameFormat (which gets called in namespec). stringConNames takes in the pattern and processes the naming. The names are coming out appropriately named so I think we are good here.